tests/trainer/connectors/test_data_connector.py

```diff
@@ -123,7 +124,6 @@ def test_override_both_model_and_datamodule(self, hook_name):
         with pytest.warns(UserWarning, match=f"have overridden `{hook_name}` in both"):
             hook = trainer._data_connector._datahook_selector.get_hook(hook_name)
 
-        warning_cache.clear()
```

Comment:
Hmm, do you know why was this present before? And if you could share why this is removed?

Reply:
it's not removed. If have moved up so that it will be refreshed with each test.